src/app/destiny1/loadout-drawer/D1LoadoutDrawer.tsx

```diff
@@ -187,42 +127,113 @@ export default function D1LoadoutDrawer({
   );
 
   return (
-    <Sheet onClose={onClose} header={header} footer={footer} disabled={showingItemPicker}>
-      <div className="loadout-drawer loadout-create">
-        <div className="loadout-content">
-          <LoadoutDrawerDropTarget onDroppedItem={onAddItem} classType={loadout.classType}>
-            {warnitems.length > 0 && (
-              <div className="loadout-contents">
-                <p>
-                  <AlertIcon />
-                  {t('Loadouts.VendorsCannotEquip')}
-                </p>
-                <div className="loadout-warn-items">
-                  {warnitems.map((li) => (
-                    <div key={li.item.id} className="loadout-item" onClick={() => fixWarnItem(li)}>
-                      <ClosableContainer onClose={(e) => onRemoveItem(li, e)}>
-                        <ItemIcon item={li.item} />
-                      </ClosableContainer>
-                    </div>
-                  ))}
-                </div>
-              </div>
-            )}
+    <Sheet onClose={onClose} header={header} footer={footer}>
+      <LoadoutDrawerBody loadout={loadout} storeId={storeId} setLoadout={setLoadout} />
+    </Sheet>
+  );
+}
+
+// This is mostly separated out so that its use of useItemPicker is under the Sheet in the component tree.
+function LoadoutDrawerBody({
+  loadout,
+  storeId,
+  setLoadout,
+}: {
+  loadout: Loadout;
+  storeId: string;
+  setLoadout: Dispatch<SetStateAction<Loadout>>;
+}) {
+  const defs = useD1Definitions()!;
+  const showItemPicker = useItemPicker();
+
+  const allItems = useSelector(allItemsSelector);
+  const itemCreationContext = useSelector(createItemContextSelector);
+
+  const loadoutItems = loadout?.items;
+
+  // Turn loadout items into real DimItems
+  const [items, warnitems] = useMemo(
+    () =>
+      getItemsFromLoadoutItems(
+        itemCreationContext,
+        loadoutItems,
+        storeId,
+        allItems,
+        undefined,
+        defs
+      ),
+    [itemCreationContext, loadoutItems, storeId, allItems, defs]
+  );
+
+  const onAddItem = useCallback(
+    (item: DimItem, equip?: boolean) => setLoadout(addItem(defs, item, equip)),
+    [defs, setLoadout]
+  );
+
+  /**
+   * If an item comes in on the addItem$ observable, add it.
+   */
```

Comment:
I know its copy paste but

```suggestion
   // If an item comes in on the addItem$ observable, add it.
```